File: backend/src/main/java/com/realdolmen/backend/model/Book.java
package com.realdolmen.backend.model;

import lombok.Data;

import javax.persistence.*;

@Entity
@Data
@Inheritance(strategy = InheritanceType.JOINED)
@DiscriminatorColumn(name = "book_type")
@DiscriminatorValue(value = "book")
public abstract class Book extends Article
{

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column
    private Long id;

    private String author;
    private String isbn;
    private int nrPages;

}

File: frontend/src/app/models/book.ts
import {Article} from "./article";

export abstract class Book extends Article {

  public author: string;
  public isbn: string;
  public nrOfPages: number;
  public bookType: string;

  constructor(author?: string, isbn?: string, nrOfPages?: number, bookType?: string) {
    super();
    this.author = author;
    this.isbn = isbn;
    this.nrOfPages = nrOfPages;
    this.bookType = bookType;
  }
}
